spec/feature/posts_show_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+RSpec.describe 'Post show page', type: :feature do
+  let!(:user) { User.create(name: 'Sarkodie', photo: '/path/to/photo1.jpg', post_counter: 10) }
+  let!(:post) do
+    Post.create(
+      title: 'Test Post',
+      text: 'This is a test post.',
+      author_id: user.id,
+      comments_counter: 3,
+      likes_counter: 7
+    )
+  end
+
+  describe 'GET #show' do
+    before { visit user_post_path(post) }
+
+    it 'displays the post title' do
+      expect(page).to have_content(post.title)
+    end
+
+    it 'displays the number of comments' do
+      expect(page).to have_content("Comments: #{post.comments_counter}")
+    end
+
+    it 'displays the number of likes' do
+      expect(page).to have_content("Likes: #{post.likes_counter}")
+    end
+
+    it 'displays the post body' do
+      expect(page).to have_content(post.text)
+    end
+
+    it 'displays the username of each commenter and their comment' do
+      post.comments.each do |comment|
+        expect(page).to have_content("#{comment.user.name} - #{comment.text}")
+      end
+    end
+  end
+end
```

Comment:
- [ ] Great that you have added test for the comment each commentor left but kindly consider to add test for the username who wrote the post. because right now you only have a test for the title of the post.

Reply:
Hi @Zuhaib042 

On the posts#show page, there is no username for the person who wrote the post. we display the post title, post text, and username for the user who left a comment, comment.text etc... check the screenshot below.

![Cap](https://github.com/CaptainBawa/Blog-App/assets/82339317/50b66755-247f-4f64-b157-8375628d967e)

For the username for the commenter, am already testing it.

![C](https://github.com/CaptainBawa/Blog-App/assets/82339317/0dd0c679-160f-4ad7-897c-fd4db2d121f1)